pandas/tests/apply/test_series_apply.py

```diff
@@ -613,7 +630,7 @@ def test_map_defaultdict_ignore_na():
     mapping = defaultdict(int, {1: 10, np.nan: 42})
     ser = Series([1, np.nan, 2])
     result = ser.map(mapping)
-    expected = Series([10, 0, 0])
+    expected = Series([10, 42, 0])
```

Comment:
According to the discussion we had in #47585 the old behavior is correct

Reply:
ahh I see. 
Do we keep it as it is now ? or is this enhancement fine - since there are some discrepancies observed (based on dtype) in map behavior as highlighted in bug description https://github.com/pandas-dev/pandas/issues/48813
TIA

Reply:
cc @rhshadrach can you weigh in?

Reply:
+1 on changing the behavior in this test. The 0 in question here is because `np.nan` does not equal itself, and NumPy often returns views so that ids are not equal either; e.g.

```
mapping = defaultdict(int, {1: 10, np.nan: 42})
arr = np.array([1, np.nan, 2])
print(mapping[arr[1]])

# 0
```

I think introducing a better lookup for NaN values makes sense, and brings this in line with the Series case:

```
mapping = pd.Series({1: 10, np.nan: 42})
ser = Series([1, np.nan, 2])
print(ser.map(mapping))

# 0    10.0
# 1    42.0
# 2     NaN
# dtype: float64
```